ngraph/test/util/engine/ie_cpu_engine.cpp

```diff
@@ -0,0 +1,95 @@
+#include "ie_cpu_engine.hpp"
+
+#include "ngraph/op/get_output_element.hpp"
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/pass/manager.hpp"
+#include "ngraph/pass/opset1_upgrade.hpp"
+
+using namespace ngraph;
+
+test::IE_CPU_Engine::IE_CPU_Engine(const std::shared_ptr<Function> function)
+    : m_function{function}
+{
+    upgrade_and_validate_function(m_function);
+    const auto cnn_network = InferenceEngine::CNNNetwork(m_function);
+    m_network_inputs = cnn_network.getInputsInfo();
+    m_network_outputs = cnn_network.getOutputsInfo();
+
+    InferenceEngine::Core ie;
+    auto exe_network = ie.LoadNetwork(cnn_network, "CPU");
+    m_inference_req = exe_network.CreateInferRequest();
+}
+
+void test::IE_CPU_Engine::infer()
+{
+    if (m_network_inputs.size() != m_allocated_inputs)
+    {
+        THROW_IE_EXCEPTION << "The tested graph has " << m_network_inputs.size() << " inputs, but "
+                           << m_allocated_inputs << " were passed.";
+    }
+    else
+    {
+        m_inference_req.Infer();
+    }
+}
+
+testing::AssertionResult test::IE_CPU_Engine::compare_results(const size_t tolerance_bits)
+{
+    auto comparison_result = testing::AssertionSuccess();
+
+    for (const auto output : m_network_outputs)
+    {
+        InferenceEngine::MemoryBlob::CPtr computed_output_blob =
+            InferenceEngine::as<InferenceEngine::MemoryBlob>(m_inference_req.GetBlob(output.first));
+
+        const auto& expected_output_blob = m_expected_outputs[output.first];
+
+        comparison_result =
+            compare_blobs(computed_output_blob, expected_output_blob, tolerance_bits);
+
+        if (comparison_result == testing::AssertionFailure())
+        {
+            break;
+        }
+    }
+
+    return comparison_result;
+}
+
+std::shared_ptr<Function> test::IE_CPU_Engine::upgrade_and_validate_function(
+    const std::shared_ptr<Function> function) const
+{
+    pass::Manager passes;
+    passes.register_pass<pass::Opset1Upgrade>();
```

Comment:
It is not topic for this task, but I think that we shouldn't use `Opset1Upgrade`

Reply:
This pass is also used in IE_Backend. This is to make sure that all unit tests (those containing opset0 ops) work with IE.